poc/tests/test_expressions.py

```diff
@@ -0,0 +1,668 @@
+import copy
+import pickle
+import unittest
+
+from poc.expressions import trait, recursive, metadata
+from poc.observe import (
+    _is_not_none,
```

Comment:
Similarly, this is available in `traits.trait_base`, as `not_none`.

Reply:
It's a bit bike-sheddy, but using a simple `lambda v: v is not None` seems more readable to me - it's understandable from the immediate context (so the reader has no need to look up the function to check that it's doing what they think it is). The `not_none` in `traits.trait_base` feels like clutter.